src/MarkdownTextInput.web.tsx

```diff
@@ -369,20 +371,54 @@ const MarkdownTextInput = React.forwardRef<TextInput, MarkdownTextInputProps>(
             text = parseText(divRef.current, changedText, processedMarkdownStyle).text;
         }
 
+        const normalizedText = normalizeValue(text);
+
         if (pasteRef?.current) {
           pasteRef.current = false;
           updateSelection(e);
         }
         updateTextColor(divRef.current, text);
 
         if (onChange) {
-          const event = e as unknown as NativeSyntheticEvent<any>;
+          const event = e as unknown as NativeSyntheticEvent<{
+            count: number;
+            before: number;
+            start: number;
+          }>;
           setEventProps(event);
+
+          const newSelection = CursorUtils.getCurrentCursorPosition(divRef.current) ?? {start: 0, end: 0};
+
+          // The new text is between the prev start selection and the new end selection
+          const maybeAddedtext = normalizedText.slice(prevSelection.start, newSelection.end);
+          // The length of the text that replaced the before text
+          const count = maybeAddedtext.length;
+          // The start index of the replacement operation
+          let start = prevSelection.start;
+
+          const prevSelectionRange = prevSelection.end - prevSelection.start;
+          // The length the deleted text had before
+          let before = prevSelectionRange;
+          if (prevSelectionRange === 0 && (nativeEvent.inputType === 'deleteContentBackward' || nativeEvent.inputType === 'deleteContentForward')) {
```

Comment:
Since now we are using `nativeEvent.inputType` quite often, let's replace `nativeEvent` variable with `inputType`